torch/testing/_internal/schema_check_tensor.py

```diff
@@ -38,16 +43,69 @@ def __repr__(self):
     @classmethod
     def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         def unwrap(e):
-            return e.elem if isinstance(e, cls) else e
+            if isinstance(e, cls):
+                return e.elem
+            elif is_iterable_of_dtype(e, cls):
```

Comment:
Isn't `tree_map` supposed to take care of this case and recursively traverse lists and dicts?